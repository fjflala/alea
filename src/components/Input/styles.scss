.ui-input {
  height: 48px;
  font-size: 16px;
  width: 100%;
  border-radius: 4px;
  margin: 8px 0;
  padding: 8px;
  border: 1px solid #ccc;
}