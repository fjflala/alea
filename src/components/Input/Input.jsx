/**
 * Module dependecies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Import styles
 */
import './styles.scss';

/**
 * Represents an Input
 * @param {object} props - Dynamic list of props based on html input properties
 * @param {string} props.className - Custom className for the element
 */
const Input = (props) => {
  const {
    className,
  } = props;

  const cn = `ui-input ${className && className}`;

  return (
    <input
      {...props}
      className={cn}
    />
  );
};

/**
 * PropTypes definitions
 */
Input.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
};

Input.defaultProps = {
  className: '',
  type: 'text',
  onChange: () => {},
  onFocus: () => {},
};

/**
 * Expose component
 */
export default Input;
