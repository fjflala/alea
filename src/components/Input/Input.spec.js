import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('<Input />', () => {
  it('should correctly render', () => {
    const props = {
      className: 'custom-class',
      onChange: jest.fn(),
      onFocus: jest.fn(),
    };

    const tree = renderer
      .create(<Input {...props} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});