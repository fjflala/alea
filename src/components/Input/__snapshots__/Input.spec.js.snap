// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should correctly render 1`] = `
<input
  className="ui-input custom-class"
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  type="text"
/>
`;
